### PR TITLE
Framework reference restrictions for plugins

### PR DESCRIPTION
**Pull request by vitek-karas**
Add a section which describes current restrictions on plugins, that they are not allowed to reference frameworks which are not also references by the host app.

Reacts to https://github.com/dotnet/core-setup/issues/8682

### CODE REVIEW

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 1]
```diff
@@ -280,3 +280,7 @@ This prevents the `A.PluginBase` assemblies from being copied to the output dire
 ## Plugin target framework recommendations
 
 Because plugin dependency loading uses the *.deps.json* file, there is a gotcha related to the plugin's target framework. Specifically, your plugins should target a runtime, such as .NET Core 3.0, instead of a version of .NET Standard. The *.deps.json* file is generated based on which framework the project targets, and since many .NET Standard-compatible packages ship reference assemblies for building against .NET Standard and implementation assemblies for specific runtimes, the *.deps.json* may not correctly see implementation assemblies, or it may grab the .NET Standard version of an assembly instead of the .NET Core version you expect.
+
+## Plugin framework references
+
+Currently plugins can not introduce new frameworks into the process. So for example trying to load a plugin which uses ASP.NET (`Microsoft.AspNetCore.App` framework) into an application which only uses the root `Microsoft.NETCore.App` framework is not allowed. The host application must declare references to all frameworks it itself and all potential plugins might need.
```
Comment (tdykstra) [inline docs/core/tutorials/creating-app-with-plugin-support.md]:
```suggestion
Currently, plugins can't introduce new frameworks into the process. So for example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.
```

**Review by tdykstra — COMMENTED** [review]
Suggested some wording changes.

**Review by tdykstra — COMMENTED** [review]
[Extra review caused by GitHub 500 error]

**Review by tdykstra — COMMENTED** [review]
[Extra review caused by GitHub 500 error]

File: docs/core/tutorials/creating-app-with-plugin-support.md [hunk 2]
```diff
@@ -280,3 +280,7 @@ This prevents the `A.PluginBase` assemblies from being copied to the output dire
 ## Plugin target framework recommendations
 
 Because plugin dependency loading uses the *.deps.json* file, there is a gotcha related to the plugin's target framework. Specifically, your plugins should target a runtime, such as .NET Core 3.0, instead of a version of .NET Standard. The *.deps.json* file is generated based on which framework the project targets, and since many .NET Standard-compatible packages ship reference assemblies for building against .NET Standard and implementation assemblies for specific runtimes, the *.deps.json* may not correctly see implementation assemblies, or it may grab the .NET Standard version of an assembly instead of the .NET Core version you expect.
+
+## Plugin framework references
+
+Currently, plugins can't introduce new frameworks into the process. So for example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.
```
Comment (elinor-fung) [inline docs/core/tutorials/creating-app-with-plugin-support.md]:
```suggestion
Currently, plugins can't introduce new frameworks into the process. For example, you can't load a plugin that uses the `Microsoft.AspNetCore.App` framework into an application that only uses the root `Microsoft.NETCore.App` framework. The host application must declare references to all frameworks needed by plugins.
```

**Review by elinor-fung — APPROVED** [review]

**Review by tdykstra — APPROVED** [review]

**Pull request merged**